ClearentIdtechIOSFramework/ClearentUI/ClearentUIManager.swift

```diff
@@ -64,24 +66,33 @@ public final class ClearentUIManager: NSObject {
      * @param amount, the amount to be charged in a transaction
      * @param completion, a closure to be executed once the clearent SDK UI is dimissed
      */
-    public func paymentViewController(amount: Double, completion: ((CompletionResult) -> Void)?) -> UINavigationController {
-        viewController(processType: .payment, amount: amount, dismissCompletion: completion)
+    @objc public func paymentViewController(amount: Double, completion: ((ClearentResult) -> Void)?) -> UINavigationController {
+        viewController(processType: .payment, dismissCompletion: { result in
+            let result =  self.resultFor(completionResult: result)
+            completion?(result)
+        })
```

Comment:
```suggestion
        viewController(processType: .payment, dismissCompletion: { [weak self] result in
            let result =  self?.resultFor(completionResult: result)
            completion?(result)
        })
```
same for the other flows